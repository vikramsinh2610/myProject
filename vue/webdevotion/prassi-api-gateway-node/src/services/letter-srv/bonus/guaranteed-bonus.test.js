const GuaranteedBonus = require('./guaranteed-bonus');

describe('GuaranteedBonus', () => {
  test('should have these properties', () => {
    const bonus = new GuaranteedBonus({
      amount: 1000,
      paymentTime: 'prepayment',
      absorbability: {
        directProductionPercentage: 0,
        indirectProductionPercentage: 100,
      },
    });
    expect(bonus).toHaveProperty('amount');
    expect(bonus).toHaveProperty('paymentTime');
    expect(bonus.absorbability).toHaveProperty('directProductionPercentage');
    expect(bonus.absorbability).toHaveProperty('indirectProductionPercentage');
  });

  test('should return it s JSON Schema', () => {
    expect(GuaranteedBonus.getJSONSchema()).toMatchSnapshot();
  });
});
