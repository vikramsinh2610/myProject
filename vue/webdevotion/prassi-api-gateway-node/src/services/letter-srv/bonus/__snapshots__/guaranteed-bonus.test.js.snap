// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuaranteedBonus should return it s JSON Schema 1`] = `
Object {
  "description": "Guaranteed Bonus",
  "properties": Object {
    "absorbability": Object {
      "properties": Object {
        "directProductionPercentage": Object {
          "description": "Direct production percentage, with 2 decimals as integer",
          "maximum": 10000,
          "minimum": 0,
          "type": "integer",
        },
        "indirectProductionPercentage": Object {
          "description": "Indirect production percentage, with 2 decimals as integer",
          "maximum": 10000,
          "minimum": 0,
          "type": "integer",
        },
      },
      "required": Array [
        "directProductionPercentage",
        "indirectProductionPercentage",
      ],
      "type": "object",
    },
    "amount": Object {
      "type": "integer",
    },
    "paymentTime": Object {
      "enum": Array [
        "prepayment",
        "payment-delayed",
      ],
      "type": "string",
    },
    "type": Object {
      "enum": Array [
        "guaranteed",
      ],
      "type": "string",
    },
  },
  "required": Array [
    "amount",
    "type",
    "paymentTime",
    "absorbability",
  ],
  "type": "object",
}
`;
